lib/archethic/bootstrap.ex

```diff
@@ -206,12 +208,93 @@ defmodule Archethic.Bootstrap do
       Logger.info("Synchronization finished")
     end
 
+    nodes = P2P.authorized_and_available_nodes()
+    # mint reward resync req to load gen addr from db
+    resync_network_chain([:oracle, :node_shared_secrets, :mint_rewards], nodes)
+
     Sync.publish_end_of_sync()
     SelfRepair.start_scheduler()
-
     :persistent_term.put(:archethic_up, :up)
   end
 
+  # when no node loaded in memtable
+  @spec resync_network_chain(any, maybe_improper_list) :: :ok
+  def resync_network_chain(_type_list, _nodes = []),
+    do: Logger.info("Enforced Resync of Network Txs: failure, No-Nodes")
+
+  def resync_network_chain(type_list, nodes) when is_list(nodes) do
+    # spawn process for each type of txn to sync
+    Task.Supervisor.async_stream_nolink(Archethic.TaskSupervisor, type_list, fn type ->
+      # acquire a(any) address of type txn, (for querying DB)
+      case TransactionChain.list_addresses_by_type(type) |> Enum.at(0) do
```

Comment:
Here I think we can reuse the persistent genesis address from PR #495.
So it will not load 2 times the `list_addresses_by_type` function

Reply:
true